src/main/ruby/truffleruby/core/truffle/time_operations.rb

```diff
@@ -91,5 +91,29 @@ def self.compose(time_class, utc_offset, p1, p2 = nil, p3 = nil, p4 = nil, p5 =
 
       Primitive.time_s_from_array(time_class, sec, min, hour, mday, month, year, nsec, is_dst, is_utc, utc_offset)
     end
+
+    def self.new_from_string(time_class, str, **options)
+      raise ArgumentError, 'time string should have ASCII compatible encoding' unless str.encoding.ascii_compatible?
+
+      # Fast path for well-formed strings.
+      # Specify the acceptable range for each component in the regexp so that if
+      # any value is out of range the match will fail and fall through.
+      if match = str.match(/\A(\d{4,5})(?:-(0[0-9]|1[012])-(0[0-9]|[12][0-9]|3[01])[ T]([01][0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9]|60)(?:\.(\d+))?\s*(\S+)?)?\z/)
```

Comment:
Do we need to validate the range or each argument here already?
I'm thinking `compose`/`Time.new` validates it already e.g. for:
```
irb(main):012> RUBY_DESCRIPTION
=> "ruby 3.3.5 (2024-09-03 revision ef084cc8f4) [x86_64-linux]"
> Time.new(2024, 13)
<internal:timev>:413:in `initialize': mon out of range (ArgumentError)
> Time.new("2024-13-01T01:02:03")
<internal:timev>:409:in `initialize': mon out of range (ArgumentError)
```

Reply:
I did need that while matching the error messages, but now that we've relaxed the specs this is no longer necessary.